cmd/snap/keymgr_test.go

```diff
@@ -0,0 +1,59 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main_test
+
+import (
+	"os"
+
+	"gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/asserts"
+	snap "github.com/snapcore/snapd/cmd/snap"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type keymgrSuite struct{}
+
+var _ = check.Suite(&keymgrSuite{})
+
+func (keymgrSuite) TestGPGKeypairManager(c *check.C) {
+	keypairMgr, err := snap.GetKeypairManager()
+	c.Check(err, check.IsNil)
+	c.Check(keypairMgr, check.FitsTypeOf, &asserts.GPGKeypairManager{})
+}
+
+func (keymgrSuite) TestExternalKeypairManager(c *check.C) {
```

Comment:
(nitpick) maybe a test for the failure case here? i.e. if the external keymgr returns "exit 1" or something to see that the error looks sensible and the code is DTRT?